modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartTransitionResourceImpl.java

```diff
@@ -82,6 +98,42 @@ public Page<CartTransition> getCartCartTransitionsPage(Long cartId)
 			else if (commerceOrder.isDraft()) {
 				transitionOVPs.add(new ObjectValuePair<>(0L, "submit"));
 			}
+
+			List<CommerceOrderItem> commerceOrderItems =
+				commerceOrder.getCommerceOrderItems();
+
+			if ((commerceOrder.getBillingAddressId() > 0) &&
+				Validator.isNotNull(
+					commerceOrder.getCommercePaymentMethodKey()) &&
+				(commerceOrder.getShippingAddressId() > 0) &&
+				(commerceOrder.getCommerceShippingMethodId() > 0) &&
```

Comment:
It doesn't matter, what we are trying to achieve here is to check out an order quickly, so we only need to check if everything is set correctly. The shipping process is something that will happen later.

Reply:
ok, but if the items are virtual I don't think the order will have shipping address id > 0 and a shipping method id > 0, right?

Reply:
This is a good question. I don't know how the shipping stuff for the virtual items works. I'll check it, thank you!

Reply:
You were right, I added the check on the shippable because the virtual items don't require the shipping fields to have a value. Resolving this one, thank you!